packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -91,6 +95,13 @@ export default function VisualEditor( { styles } ) {
 				// We should auto-focus the canvas (title) on load.
 				// eslint-disable-next-line jsx-a11y/no-autofocus
 				autoFocus={ ! isWelcomeGuideVisible }
+				iframeProps={ {
+					style: {
+						background: isEditingPattern
+							? 'transparent'
```

Comment:
This is the magic that removes the background and only highlights the pattern being edited. This is current behaviour in trunk.

To be effective however, it relies on the pattern having a background of its own. 

https://github.com/WordPress/gutenberg/assets/6458278/4e5a9bb5-3d8b-46be-8d82-7e3e9e740c1d




cc @glendaviesnz for confirmation

Reply:
Actually I'm thinking we ditch the transparent background.